frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -227,32 +275,117 @@ function BookInfo:onShowBookDescription(description, file)
-            document:loadDocument(false) -- load only metadata
-        end
+function BookInfo:onShowBookCover(file, force_orig)
+    local cover_bb = self:getCoverPageImage(self.document, file, force_orig)
+    if cover_bb then
+        local imgviewer = ImageViewer:new{
+            image = cover_bb,
+            with_title_bar = false,
+            fullscreen = true,
+        }
+        UIManager:show(imgviewer)
     else
-        document = self.document
+        UIManager:show(InfoMessage:new{
+            text = _("No cover image available."),
+        })
     end
-    if document then
-        local cover_bb = document:getCoverPageImage()
-        if cover_bb then
-            local ImageViewer = require("ui/widget/imageviewer")
-            local imgviewer = ImageViewer:new{
-                image = cover_bb,
-                with_title_bar = false,
-                fullscreen = true,
-            }
-            UIManager:show(imgviewer)
+end
+
+function BookInfo:getCoverPageImage(doc, file, force_orig)
+    local cover_bb
+    local custom_cover = self:getCustomBookCover(file or (doc and doc.file))
+    if not force_orig and custom_cover then
+        local img_widget = ImageWidget:new{
+            file = custom_cover,
+        }
+        cover_bb = img_widget:getImage()
+        img_widget:free()
+    else
+        local not_is_doc = not doc and true or false
+        if not_is_doc then
+            doc = DocumentRegistry:openDocument(file)
+            if doc and doc.loadDocument then -- CreDocument
+                doc:loadDocument(false) -- load only metadata
+            end
+        end
+        if doc then
+            cover_bb = doc:getCoverPageImage()
+            if not_is_doc then
+                doc:close()
+            end
+        end
+    end
+    return cover_bb
+end
+
+function BookInfo:getCustomBookCover(file)
+    local sidecar_dir = DocSettings:getSidecarDir(file, "doc")
+    local cover_file = DocSettings:getCoverFile(sidecar_dir)
+    if not cover_file then
+        sidecar_dir = DocSettings:getSidecarDir(file, "dir")
+        cover_file = DocSettings:getCoverFile(sidecar_dir)
+    end
+    return cover_file
+end
+
+function BookInfo:setCustomBookCover(file, book_props, caller_callback, cover_file)
+    local function kvp_update()
+        self.updated = true
+        self.kvp_widget:onClose()
+        if self.document then
+            self:onShowBookInfo()
         else
-            UIManager:show(InfoMessage:new{
-                text = _("No cover image available."),
-            })
+            self:show(file, book_props, caller_callback)
         end
-        document:close()
+    end
+    if cover_file then -- reset custom cover
+        local ConfirmBox = require("ui/widget/confirmbox")
+        local confirm_box = ConfirmBox:new{
+            text = _("Reset custom cover?"),
+            ok_text = _("Reset"),
+            ok_callback = function()
+                if os.remove(cover_file) then
```

Comment:
I'm not sure I'd necessarily expect this as a user who put a cover.jpg or something in there. Though on the other hand, what else could it do. Maybe an extra line in the text saying it'll be removed similar to .sdr?